R/learner_lightgbm_regr_lightgbm.R

```diff
@@ -223,9 +231,95 @@ LearnerRegrLightGBM = R6Class("LearnerRegrLightGBM",
   ),
 
   private = list(
+    .validate = NULL,
+
     .train = function(task) {
       pars = self$param_set$get_values(tags = "train")
-      train_lightgbm(self, task, "regr", pars)
+
+      # convert data
+      x_train = data.matrix(task$data(rows = task$row_roles$use, cols = task$feature_names))
+      y_train = task$data(rows = task$row_roles$use, cols = task$target_names)[[1L]]
+
+      # create data set
+      categorical_feature = if (any(task$feature_types$type %in% c("factor", "logical"))) {
+        task$feature_types$id[task$feature_types$type %in% c("factor", "logical")]
+      }
+
+      dtrain = lightgbm::lgb.Dataset(
+        data = x_train,
+        label = y_train,
+        free_raw_data = FALSE,
+        categorical_feature = categorical_feature
+      )
+      if ("weights" %in% task$properties) {
+        dtrain$set_field("weight", task$weights[get("row_id") %in% task$row_roles$use, "weight"][[1L]])
```

Comment:
is it ensured that `task$weigths` returns them sorted by row_id? Otherwise the weights might be differently ordered than x_train and y_train

Reply:
Done